MPU/include/driverio.h

```diff
@@ -28,50 +29,20 @@
 class DRIVERIO
 {
     private:
-
         CASCADIAMC *motorController;
         ORIONBMS *bms;
 
-        Timer ssButton_debounce;
-        Timer powerToggle_wait;
-        Timer speaker_wait;
-        Timer tempWarningBlink_wait;
-
-        bool LED5_status = false;
-
-        /**
-         * @brief Checks if the button is being pressed and the timer isn't canceled and the timer isn't canceled
-         * 
-         */
-        bool ssButtonDebounced();
-
-        /**
-         * @brief Writes LED4 to a specific state
-         * 
-         * @param state 
-         */
-        void writeLED4(bool state);
+        SPEAKER speaker;
+        LED socLED;
+        LED tempLED;
+        LED yLED;
```

Comment:
Might be cleaner to get rid of the default constructor in all of the hardware classes and only have the one that initializes a pin value. Doesn't really make sense to have an LED or SWITCH without a pin value, so there shouldn't be a way to construct an object like that. You could make these initializations look like `LED yLED {YLED_PIN};`, and that also makes it so you don't have to reinitialize them in the DRIVERIO constructor (reduces code and seems cleaner).

Reply:
I did not know you could do that, thank you thank you I will use this so much

Reply:
Done for the DriverIO HW, I will do it for like everything now lol

Reply:
I'm a little bit rusty on my C++, so the more I think about this I'm a little unsure if it's valid. The fact that it initializes variables in the header file could be an issue. If it fails you can probably make them `static const` variables, but just keep it in mind when you're testing and make sure it compiles.

Reply:
I mean it compiled soooo

Reply:
Yeah I think peter is totally right. That should be the way it's done and should be cleaner